exporters/otlp/otlplog/otlploghttp/client_test.go

```diff
@@ -700,6 +700,33 @@ func TestConfig(t *testing.T) {
 		assert.Len(t, rCh, 0, "failed HTTP responses did not occur")
 	})
 
+	t.Run("WithRetryAndExporterErr", func(t *testing.T) {
+		exporterErr := errors.New("rpc error: code = Unavailable desc = service.name not found in resource attributes")
+		rCh := make(chan exportResult, 1)
+		rCh <- exportResult{Err: &httpResponseError{
+			Status: http.StatusTooManyRequests,
+			Err:    exporterErr,
+		}}
+		exp, coll := factoryFunc("", rCh, WithRetry(RetryConfig{
+			Enabled: false,
+		}))
+		ctx := context.Background()
+		t.Cleanup(func() { require.NoError(t, coll.Shutdown(ctx)) })
+		// Push this after Shutdown so the HTTP server doesn't hang.
+		t.Cleanup(func() { close(rCh) })
+		t.Cleanup(func() { require.NoError(t, exp.Shutdown(ctx)) })
+		err := exp.Export(ctx, make([]log.Record, 1))
+		assert.Error(t, err, exporterErr)
+
+		var retryErr *retryableError
+		assert.True(t, errors.As(err, &retryErr))
+		assert.Equal(t, fmt.Sprintf("429: %v", exporterErr), retryErr.Unwrap().Error())
+
+		var newErr *retryableError
+		assert.True(t, retryErr.As(&newErr))
+		assert.Equal(t, newErr.Unwrap().Error(), retryErr.Unwrap().Error())
```

Comment:
This seems like it is testing the wrapping semantics of Go itself by checking multiple unwraps here. Is that correct?

If so, I think the assertion above `assert.ErrorIs` should be sufficient. It calls `Unwrap` recursively until the base `exportErr` is returned and should be sufficient to validate the `Unwrap` implementation.

Reply:
Basically, wanted to test the `As` implementation and then verify the error. Changed it to `ErrorIs`.